src/webgl/camera.js

```diff
@@ -78,14 +182,33 @@ p5.prototype.camera = function(x, y, z){
  * colored 3d boxes toggleable with mouse position
  *
  */
-p5.prototype.perspective = function(fovy,aspect,near,far) {
-  fovy = fovy || (60 / 180 * this.PI);
-  aspect = aspect || (this.width/this.height);
-  near = near || ((this.height/2.0) / this.tan(fovy/2.0) * 0.1);
-  far = far || ((this.height/2.0) / this.tan(fovy/2.0) * 10);
-  this._renderer.uPMatrix = p5.Matrix.identity();
-  this._renderer.uPMatrix.perspective(fovy,aspect,near,far);
-  this._renderer._curCamera = 'custom';
+p5.prototype.perspective = function() {
+  this._renderer.perspective.apply(this._renderer, arguments);
+};
+
+p5.RendererGL.prototype.perspective = function() {
+  var fovy = arguments[0] || this.defaultCameraFOV;
+  var aspect = arguments[1] || this.defaultCameraAspect;
+  var near = arguments[2] || this.defaultCameraNear;
+  var far = arguments[3] || this.defaultCameraFar;
```

Comment:
are all four of these optional, or is this just a safeguard? if optional, let's indicate this in the docs with [] around the params